src/mapnode.cpp

```diff
@@ -601,14 +601,21 @@ Buffer<u8> MapNode::serializeBulk(int version,
 
 	// Writing to the buffer linearly is faster
 	u8 *p = &databuf[0];
-	for (u32 i = 0; i < nodecount; i++, p += 2)
-		writeU16(p, nodes[i].param0);
-
-	for (u32 i = 0; i < nodecount; i++, p++)
-		writeU8(p, nodes[i].param1);
-
-	for (u32 i = 0; i < nodecount; i++, p++)
-		writeU8(p, nodes[i].param2);
+	if (is_mono_block) {
+		for (u32 i = 0; i < nodecount; i++, p += 2)
```

Comment:
nitpick: could make sense to make a copy of `nodes[0]` in a variable, compiler might not realize that it can't change